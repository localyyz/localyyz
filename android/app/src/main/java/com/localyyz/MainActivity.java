package com.localyyz;

import android.content.Intent;
import android.os.Bundle;

import com.facebook.react.ReactActivity;
import com.facebook.react.ReactActivityDelegate;

import javax.annotation.Nullable;

import io.branch.rnbranch.RNBranchModule;

public class MainActivity extends ReactActivity {

    /**
     * Returns the name of the main component registered from JavaScript.
     * This is used to schedule rendering of the component.
     */
    @Override
    protected String getMainComponentName() {
        return "localyyz";
    }

    @Override
    protected void onStart() {
        super.onStart();
        RNBranchModule.initSession(getIntent().getData(), this);
    }

    @Override
    public void onNewIntent(Intent intent) {
        setIntent(intent);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        MainApplication.getCallbackManager().onActivityResult(requestCode, resultCode, data);
    }

    @Override
    protected ReactActivityDelegate createReactActivityDelegate() {
        return new ReactActivityDelegate(this, getMainComponentName()) {
            @Nullable
            @Override
            protected Bundle getLaunchOptions() {
                Bundle initialProps = new Bundle();
                initialProps.putString("API_URL", BuildConfig.API_URL);
                return initialProps;
            }
        };
    }

}
