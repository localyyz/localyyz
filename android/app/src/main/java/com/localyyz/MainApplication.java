package com.localyyz;

import android.app.Application;

import com.BV.LinearGradient.LinearGradientPackage;
import com.cmcewen.blurview.BlurViewPackage;
import com.facebook.CallbackManager;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.react.ReactApplication;
import com.geektime.rnonesignalandroid.ReactNativeOneSignalPackage;
import com.idehub.GoogleAnalyticsBridge.GoogleAnalyticsBridgePackage;
import com.brentvatne.react.ReactVideoPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.reactnative.androidsdk.FBSDKPackage;
import com.facebook.soloader.SoLoader;
import com.learnium.RNDeviceInfo.RNDeviceInfo;
import com.microsoft.codepush.react.CodePush;
import com.oblador.vectoricons.VectorIconsPackage;
import com.reactnative.photoview.PhotoViewPackage;

import java.util.Arrays;
import java.util.List;

import ca.jaysoo.extradimensions.ExtraDimensionsPackage;
import io.branch.referral.Branch;
import io.branch.rnbranch.RNBranchPackage;

//import com.reactnativepayments.ReactNativePaymentsPackage;

public class MainApplication extends Application implements ReactApplication {

  private static CallbackManager mCallbackManager = CallbackManager.Factory.create();

  protected static CallbackManager getCallbackManager() {
    return mCallbackManager;
  }

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {

    @Override
    protected String getJSBundleFile() {
      return CodePush.getJSBundleFile();
    }
    
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new ReactNativeOneSignalPackage(),
            new GoogleAnalyticsBridgePackage(),
            new ReactVideoPackage(),
            new ExtraDimensionsPackage(),
              new CodePush(BuildConfig.CODEPUSH_KEY, getApplicationContext(), BuildConfig.DEBUG),
              new RNBranchPackage(),
              //new ReactNativePaymentsPackage(),
              new FBSDKPackage(mCallbackManager),
              new PhotoViewPackage(),
              new VectorIconsPackage(),
              new LinearGradientPackage(),
              new RNDeviceInfo(),
              new BlurViewPackage()
      );
    }

    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
      return getPackages();
    }


    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();

    FacebookSdk.sdkInitialize(getApplicationContext());
    // If you want to use AppEventsLogger to log events.
    AppEventsLogger.activateApp(this);

    // Branch
    Branch.getAutoInstance(this);

    SoLoader.init(this, /* native exopackage */ false);
  }

}
