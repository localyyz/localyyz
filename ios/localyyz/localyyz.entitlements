<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>aps-environment</key>
	<string>development</string>
	<key>com.apple.developer.associated-domains</key>
	<array>
		<string>applinks:localyyz.app.link</string>
		<string>applinks:localyyz-alternate.app.link</string>
		<string>applinks:localyyz.test-app.link</string>
		<string>applinks:localyyz-alternate.test-app.link</string>
	</array>
	<key>com.apple.developer.in-app-payments</key>
	<array>
		<string>merchant.com.localyyz.toronto</string>
	</array>
</dict>
</plist>
