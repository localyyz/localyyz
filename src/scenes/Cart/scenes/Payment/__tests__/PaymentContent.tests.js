import React from "react";

// custom
import { renderWithLayout, findById } from "localyyz/tests";

// local
import ComponentWrapper from "../content";
const Component = ComponentWrapper.wrappedComponent || ComponentWrapper;

function createProps(extra) {
  return {
    navigation: {
      navigate: jest.fn()
    },
    ...extra
  };
}

describe(Component.name, () => {
  it("should render properly to snapshot", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let [instance, layout] = [c.getInstance(), c.toJSON()];

    expect(instance).toBeDefined();
    expect(layout).toMatchSnapshot();
  });

  // test validators
  it("should validate valid card", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.validateNumberValid("4242 4242 4242 4242")).toBe(true);
    expect(instance.validateNumberValid("4242424242424242")).toBe(true);
  });

  it("should validate invalid card", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.validateNumberValid("4242 4242 4242 4244")).not.toBe(true);
    expect(instance.validateNumberValid("4242424242424244")).not.toBe(true);
    expect(instance.validateNumberValid("42424242424242")).not.toBe(true);
    expect(instance.validateNumberValid("424242424242424242")).not.toBe(true);
  });

  it("should validate valid expiry format", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.validateExpiryWellFormatted("01/01")).toBe(true);
    expect(instance.validateExpiryWellFormatted("12/99")).toBe(true);
  });

  it("should validate invalid expiry format", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.validateExpiryWellFormatted("13/01")).not.toBe(true);
    expect(instance.validateExpiryWellFormatted("5/99")).not.toBe(true);
    expect(instance.validateExpiryWellFormatted("1199")).not.toBe(true);
  });

  it("should validate valid cvc", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.validateCvcValid("111")).toBe(true);
  });

  it("should validate invalid cvc", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.validateCvcValid("11")).not.toBe(true);
    expect(instance.validateCvcValid("1111")).not.toBe(true);
    expect(instance.validateCvcValid("xxx")).not.toBe(true);
  });

  // test masks
  it("should properly mask card", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.maskCardSpaces("11111")).toBe("1111 1");
    expect(instance.maskCardSpaces("111111111")).toBe("1111 1111 1");
    expect(instance.maskCardSpaces("1111111111111")).toBe("1111 1111 1111 1");
    expect(instance.maskCardSpaces("1111111111111111")).toBe(
      "1111 1111 1111 1111"
    );
  });

  it("should properly mask expiry", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    expect(instance.maskExpirySlashes("11")).toBe("11");
    expect(instance.maskExpirySlashes("111")).toBe("11/1");
    expect(instance.maskExpirySlashes("1111")).toBe("11/11");
    expect(instance.maskExpirySlashes("11111")).toBe("11/11/1");
  });

  it("should show next button on complete card", () => {
    let props = createProps({
      card: {
        isComplete: true,
        number: "4242 4242 4242 4242",
        expiryMonth: "11",
        expiryYear: "99",
        cvc: "999"
      }
    });
    let c = renderWithLayout(<Component {...props} />, undefined, true);
    let layout = c.toJSON();

    expect(findById(layout, "next")).toBeDefined();
    expect(layout).toMatchSnapshot();
  });

  it("should not show next button on incomplete card", () => {
    let props = createProps({
      card: {
        isComplete: false,
        number: "4242 4242 4242 4242",
        expiryMonth: "11",
        expiryYear: "99"
      }
    });
    let c = renderWithLayout(<Component {...props} />, undefined, true);
    let layout = c.toJSON();

    expect(findById(layout, "next")).not.toBeDefined();
    expect(layout).toMatchSnapshot();
  });

  it("should prompt address selection for billing", () => {
    let props = createProps();
    let c = renderWithLayout(<Component {...props} />);
    let instance = c.getInstance();

    // call changeBilling like onPress
    instance.changeBilling();

    expect(props.navigation.navigate.mock.calls[0][0]).toBe("Addresses");
  });
});
