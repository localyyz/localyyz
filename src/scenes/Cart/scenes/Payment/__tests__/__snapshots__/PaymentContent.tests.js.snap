// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentContent should not show next button on incomplete card 1`] = `
<Provider
  formStore={
    FormUIStore {
      "_data": Object {
        "cvc": Object {
          "error": null,
          "id": "cvc",
          "isRequired": true,
          "mask": null,
          "onValueChange": null,
          "validators": Array [
            [Function],
            [Function],
            [Function],
          ],
          "value": null,
        },
        "expiry": Object {
          "error": null,
          "id": "expiry",
          "isRequired": true,
          "mask": [Function],
          "onValueChange": null,
          "validators": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "value": null,
        },
        "number": Object {
          "error": null,
          "id": "number",
          "isRequired": true,
          "mask": [Function],
          "onValueChange": null,
          "validators": Array [
            [Function],
            [Function],
            [Function],
          ],
          "value": null,
        },
      },
      "addCallback": [Function],
      "getError": [Function],
      "merge": [Function],
      "onValueChange": [Function],
      "update": [Function],
    }
  }
>
  <CartBaseContent>
    <Component
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <inject-Field
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          field="number"
          inputStyle={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "rgb(30, 30, 30)",
              "flex": 1,
              "fontSize": 32,
              "fontWeight": "700",
              "lineHeight": 38.4,
            }
          }
          keyboardType="numeric"
          label="Card number"
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "height": 25,
          }
        }
      />
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <inject-Field
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          field="expiry"
          inputStyle={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "rgb(30, 30, 30)",
              "flex": 1,
              "fontSize": 32,
              "fontWeight": "700",
              "lineHeight": 38.4,
            }
          }
          keyboardType="numeric"
          label="Expiry"
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        />
        <inject-Field
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          field="cvc"
          inputStyle={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "rgb(30, 30, 30)",
              "flex": 1,
              "fontSize": 32,
              "fontWeight": "700",
              "lineHeight": 38.4,
            }
          }
          keyboardType="numeric"
          label="CVV (Back 3 digits)"
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "height": 25,
          }
        }
      />
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <BaseField
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          hasError={false}
          isHorizontal={false}
          label="Billing address"
          onPress={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Select an address
          </Text>
        </BaseField>
      </Component>
    </Component>
  </CartBaseContent>
</Provider>
`;

exports[`PaymentContent should render properly to snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 874.3333333333334,
        "marginHorizontal": 15,
        "marginTop": 15,
        "padding": 15,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 150,
            },
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "paddingHorizontal": 15,
                "paddingVertical": 10,
              },
              Object {
                "paddingHorizontal": undefined,
                "paddingVertical": undefined,
              },
              false,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 15,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 14,
                      "fontWeight": "300",
                    },
                    undefined,
                  ]
                }
              >
                Card number
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                Object {
                  "flex": undefined,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              contentStyle={
                Object {
                  "flex": undefined,
                }
              }
              field="number"
              inputStyle={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "rgb(30, 30, 30)",
                  "flex": 1,
                  "fontSize": 32,
                  "fontWeight": "700",
                  "lineHeight": 38.4,
                }
              }
              isRequired={true}
              keyboardType="numeric"
              label="Card number"
              onChangeText={[Function]}
              onValueChange={[Function]}
              placeholder="required"
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 14,
                  },
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "rgb(30, 30, 30)",
                    "flex": 1,
                    "fontSize": 32,
                    "fontWeight": "700",
                    "lineHeight": 38.4,
                  },
                ]
              }
              validators={
                Array [
                  [Function],
                  [Function],
                  [Function],
                ]
              }
              value={null}
              wrapperStyle={
                Object {
                  "paddingHorizontal": undefined,
                  "paddingVertical": undefined,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 25,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 150,
            },
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "paddingHorizontal": 15,
                "paddingVertical": 10,
              },
              Object {
                "paddingHorizontal": undefined,
                "paddingVertical": undefined,
              },
              false,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 15,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 14,
                      "fontWeight": "300",
                    },
                    undefined,
                  ]
                }
              >
                Expiry
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                Object {
                  "flex": undefined,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              contentStyle={
                Object {
                  "flex": undefined,
                }
              }
              field="expiry"
              inputStyle={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "rgb(30, 30, 30)",
                  "flex": 1,
                  "fontSize": 32,
                  "fontWeight": "700",
                  "lineHeight": 38.4,
                }
              }
              isRequired={true}
              keyboardType="numeric"
              label="Expiry"
              onChangeText={[Function]}
              onValueChange={[Function]}
              placeholder="required"
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 14,
                  },
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "rgb(30, 30, 30)",
                    "flex": 1,
                    "fontSize": 32,
                    "fontWeight": "700",
                    "lineHeight": 38.4,
                  },
                ]
              }
              validators={
                Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ]
              }
              value={null}
              wrapperStyle={
                Object {
                  "paddingHorizontal": undefined,
                  "paddingVertical": undefined,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 150,
            },
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "paddingHorizontal": 15,
                "paddingVertical": 10,
              },
              Object {
                "paddingHorizontal": undefined,
                "paddingVertical": undefined,
              },
              false,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 15,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 14,
                      "fontWeight": "300",
                    },
                    undefined,
                  ]
                }
              >
                CVV (Back 3 digits)
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                Object {
                  "flex": undefined,
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              contentStyle={
                Object {
                  "flex": undefined,
                }
              }
              field="cvc"
              inputStyle={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "rgb(30, 30, 30)",
                  "flex": 1,
                  "fontSize": 32,
                  "fontWeight": "700",
                  "lineHeight": 38.4,
                }
              }
              isRequired={true}
              keyboardType="numeric"
              label="CVV (Back 3 digits)"
              onChangeText={[Function]}
              onValueChange={[Function]}
              placeholder="required"
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 14,
                  },
                  Object {
                    "backgroundColor": "rgba(255, 255, 255, 0)",
                    "color": "rgb(30, 30, 30)",
                    "flex": 1,
                    "fontSize": 32,
                    "fontWeight": "700",
                    "lineHeight": 38.4,
                  },
                ]
              }
              validators={
                Array [
                  [Function],
                  [Function],
                  [Function],
                ]
              }
              value={null}
              wrapperStyle={
                Object {
                  "paddingHorizontal": undefined,
                  "paddingVertical": undefined,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 25,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "flexBasis": 150,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "paddingHorizontal": 15,
                "paddingVertical": 10,
              },
              Object {
                "paddingHorizontal": undefined,
                "paddingVertical": undefined,
              },
              false,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 15,
              }
            }
          >
            <View
              style={Object {}}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 0)",
                      "color": "rgb(30, 30, 30)",
                      "fontSize": 14,
                      "fontWeight": "300",
                    },
                    undefined,
                  ]
                }
              >
                Billing address
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                Object {
                  "flex": undefined,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              Select an address
            </Text>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontSize": 16,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PaymentContent should show next button on complete card 1`] = `
<Provider
  formStore={
    FormUIStore {
      "_data": Object {
        "cvc": Object {
          "error": null,
          "id": "cvc",
          "isRequired": true,
          "mask": null,
          "onValueChange": null,
          "validators": Array [
            [Function],
            [Function],
            [Function],
          ],
          "value": "999",
        },
        "expiry": Object {
          "error": null,
          "id": "expiry",
          "isRequired": true,
          "mask": [Function],
          "onValueChange": null,
          "validators": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "value": "11/99",
        },
        "number": Object {
          "error": null,
          "id": "number",
          "isRequired": true,
          "mask": [Function],
          "onValueChange": null,
          "validators": Array [
            [Function],
            [Function],
            [Function],
          ],
          "value": "4242 4242 4242 4242",
        },
      },
      "addCallback": [Function],
      "getError": [Function],
      "merge": [Function],
      "onValueChange": [Function],
      "update": [Function],
    }
  }
>
  <CartBaseContent>
    <Component
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <inject-Field
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          field="number"
          inputStyle={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "rgb(30, 30, 30)",
              "flex": 1,
              "fontSize": 32,
              "fontWeight": "700",
              "lineHeight": 38.4,
            }
          }
          keyboardType="numeric"
          label="Card number"
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "height": 25,
          }
        }
      />
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <inject-Field
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          field="expiry"
          inputStyle={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "rgb(30, 30, 30)",
              "flex": 1,
              "fontSize": 32,
              "fontWeight": "700",
              "lineHeight": 38.4,
            }
          }
          keyboardType="numeric"
          label="Expiry"
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        />
        <inject-Field
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          field="cvc"
          inputStyle={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "color": "rgb(30, 30, 30)",
              "flex": 1,
              "fontSize": 32,
              "fontWeight": "700",
              "lineHeight": 38.4,
            }
          }
          keyboardType="numeric"
          label="CVV (Back 3 digits)"
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "height": 25,
          }
        }
      />
      <Component
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <BaseField
          contentStyle={
            Object {
              "flex": undefined,
            }
          }
          hasError={false}
          isHorizontal={false}
          label="Billing address"
          onPress={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          wrapperStyle={
            Object {
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "600",
              }
            }
          >
            Select an address
          </Text>
        </BaseField>
      </Component>
    </Component>
    <Component
      style={
        Object {
          "marginHorizontal": 15,
        }
      }
      testID="next"
    >
      <inject-Button
        onPress={[Function]}
      >
        Next
      </inject-Button>
    </Component>
  </CartBaseContent>
</Provider>
`;
